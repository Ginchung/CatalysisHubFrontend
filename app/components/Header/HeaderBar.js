import styled from 'styled-components';

export default styled.div`
  position: block;
  background-color: #fff;
`;

