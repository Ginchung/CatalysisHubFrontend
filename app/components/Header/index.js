import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router';
import Button from 'material-ui/Button';

import { appBar } from 'utils/constants';

import Img from './Img';
import NavBar from './NavBar';
// import HeaderLink from './HeaderLink';
import Banner from './banner.png';
import messages from './messages';
import HeaderBar from './HeaderBar';


// import DrawerBar from 'components/DrawerBar';

class Header extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <HeaderBar>
        <Link to="/">
          <Img height="100px" src={Banner} alt="react-boilerplate - Logo" />
        </Link>
        { appBar ? null :
        <NavBar
          style={{
            marginTop: '15px',
          }}
        >
          <Link to="/energies">
            <Button color="primary" >
              <FormattedMessage {...messages.energies} />
            </Button>
          </Link>


          <Link to="/generalSearch">
            <Button color="primary" >
              <FormattedMessage {...messages.generalSearch} />
            </Button>
          </Link>


          <Link to="/activityMaps">
            <Button color="primary" >
              <FormattedMessage {...messages.activityMaps} />
            </Button>
          </Link>


          <Link to="/scalingRelations">
            <Button color="primary" >
              <FormattedMessage {...messages.scalingRelations} />
            </Button>
          </Link>


          <Link to="/pourbaixDiagrams">
            <Button color="primary" >
              <FormattedMessage {...messages.pourbaixDiagrams} />
            </Button>
          </Link>


          <Link to="/publications">
            <Button color="primary" >
              <FormattedMessage {...messages.publications} />
            </Button>
          </Link>

          <Link to="/graphQLConsole">
            <Button color="primary" >
              <FormattedMessage {...messages.graphqlconsole} />
            </Button>
          </Link>

          <Link to="/yourNextApp">
            <Button color="primary" >
              <FormattedMessage {...messages.yourNextApp} />
            </Button>
          </Link>


          {/*
          <HeaderLink to="/energies" color="primary">
            <FormattedMessage {...messages.energies} />
          </HeaderLink>
          <HeaderLink to="/activityMaps">
            <FormattedMessage {...messages.activityMaps} />
          </HeaderLink>
          <HeaderLink to="/scalingRelations">
            <FormattedMessage {...messages.scalingRelations} />
          </HeaderLink>
          <HeaderLink to="/pourbaixDiagrams">
            <FormattedMessage {...messages.pourbaixDiagrams} />
          </HeaderLink>
          <HeaderLink to="/generalSearch">
            <FormattedMessage {...messages.generalSearch} />
          </HeaderLink>
          <HeaderLink to="/publications">
            <FormattedMessage {...messages.publications} />
          </HeaderLink>
          <HeaderLink to="/yourNextApp">
            <FormattedMessage {...messages.yourNextApp} />
          </HeaderLink>
          */}
        </NavBar>
        }
      </HeaderBar>
    );
  }
}

export default Header;
