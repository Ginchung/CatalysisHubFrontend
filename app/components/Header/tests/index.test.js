import React from 'react';
import { shallow } from 'enzyme';

import Header from '../index';

describe('<Header />', () => {
  it('should render a div', () => {
    const tree = shallow(
      <Header />
    );

    expect(tree).toMatchSnapshot();
  });
});
