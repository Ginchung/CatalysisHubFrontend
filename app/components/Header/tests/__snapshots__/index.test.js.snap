// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a div 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.a
      href="http://suncat.stanford.edu/"
    >
      <Styled(undefined)
        alt="react-boilerplate - Logo"
        height="100px"
        src="IMAGE_MOCK"
      />
    </styled.a>
    <styled.div
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      <Styled(Link)
        to="/energies"
      >
        <FormattedMessage
          defaultMessage="Energies"
          id="boilerplate.components.Header.energies"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
      <Styled(Link)
        to="/activityMaps"
      >
        <FormattedMessage
          defaultMessage="Activity Maps"
          id="boilerplate.components.Header.activityMaps"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
      <Styled(Link)
        to="/scalingRelations"
      >
        <FormattedMessage
          defaultMessage="Scaling Relations"
          id="boilerplate.components.Header.scalingRelations"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
      <Styled(Link)
        to="/pourbaixDiagrams"
      >
        <FormattedMessage
          defaultMessage="Pourbaix Diagrams"
          id="boilerplate.components.Header.pourbaixDiagrams"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
      <Styled(Link)
        to="/generalSearch"
      >
        <FormattedMessage
          defaultMessage="General Search"
          id="boilerplate.components.Header.generalSearch"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
      <Styled(Link)
        to="/publications"
      >
        <FormattedMessage
          defaultMessage="Publications"
          id="boilerplate.components.Header.publications"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
      <Styled(Link)
        to="/yourNextApp"
      >
        <FormattedMessage
          defaultMessage="Your Next App ..."
          id="boilerplate.components.Header.yourNextApp"
          tagName="span"
          values={Object {}}
        />
      </Styled(Link)>
    </styled.div>
  </styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.a
        href="http://suncat.stanford.edu/"
      >
        <Styled(undefined)
          alt="react-boilerplate - Logo"
          height="100px"
          src="IMAGE_MOCK"
        />
      </styled.a>
      <styled.div
        style={
          Object {
            "marginTop": "15px",
          }
        }
      >
        <Styled(Link)
          to="/energies"
        >
          <FormattedMessage
            defaultMessage="Energies"
            id="boilerplate.components.Header.energies"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
        <Styled(Link)
          to="/activityMaps"
        >
          <FormattedMessage
            defaultMessage="Activity Maps"
            id="boilerplate.components.Header.activityMaps"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
        <Styled(Link)
          to="/scalingRelations"
        >
          <FormattedMessage
            defaultMessage="Scaling Relations"
            id="boilerplate.components.Header.scalingRelations"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
        <Styled(Link)
          to="/pourbaixDiagrams"
        >
          <FormattedMessage
            defaultMessage="Pourbaix Diagrams"
            id="boilerplate.components.Header.pourbaixDiagrams"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
        <Styled(Link)
          to="/generalSearch"
        >
          <FormattedMessage
            defaultMessage="General Search"
            id="boilerplate.components.Header.generalSearch"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
        <Styled(Link)
          to="/publications"
        >
          <FormattedMessage
            defaultMessage="Publications"
            id="boilerplate.components.Header.publications"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
        <Styled(Link)
          to="/yourNextApp"
        >
          <FormattedMessage
            defaultMessage="Your Next App ..."
            id="boilerplate.components.Header.yourNextApp"
            tagName="span"
            values={Object {}}
          />
        </Styled(Link)>
      </styled.div>
    </styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.a
            href="http://suncat.stanford.edu/"
          >
            <Styled(undefined)
              alt="react-boilerplate - Logo"
              height="100px"
              src="IMAGE_MOCK"
            />
          </styled.a>
          <styled.div
            style={
              Object {
                "marginTop": "15px",
              }
            }
          >
            <Styled(Link)
              to="/energies"
            >
              <FormattedMessage
                defaultMessage="Energies"
                id="boilerplate.components.Header.energies"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/activityMaps"
            >
              <FormattedMessage
                defaultMessage="Activity Maps"
                id="boilerplate.components.Header.activityMaps"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/scalingRelations"
            >
              <FormattedMessage
                defaultMessage="Scaling Relations"
                id="boilerplate.components.Header.scalingRelations"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/pourbaixDiagrams"
            >
              <FormattedMessage
                defaultMessage="Pourbaix Diagrams"
                id="boilerplate.components.Header.pourbaixDiagrams"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/generalSearch"
            >
              <FormattedMessage
                defaultMessage="General Search"
                id="boilerplate.components.Header.generalSearch"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/publications"
            >
              <FormattedMessage
                defaultMessage="Publications"
                id="boilerplate.components.Header.publications"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/yourNextApp"
            >
              <FormattedMessage
                defaultMessage="Your Next App ..."
                id="boilerplate.components.Header.yourNextApp"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
          </styled.div>
        </styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.a
            href="http://suncat.stanford.edu/"
          >
            <Styled(undefined)
              alt="react-boilerplate - Logo"
              height="100px"
              src="IMAGE_MOCK"
            />
          </styled.a>
          <styled.div
            style={
              Object {
                "marginTop": "15px",
              }
            }
          >
            <Styled(Link)
              to="/energies"
            >
              <FormattedMessage
                defaultMessage="Energies"
                id="boilerplate.components.Header.energies"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/activityMaps"
            >
              <FormattedMessage
                defaultMessage="Activity Maps"
                id="boilerplate.components.Header.activityMaps"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/scalingRelations"
            >
              <FormattedMessage
                defaultMessage="Scaling Relations"
                id="boilerplate.components.Header.scalingRelations"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/pourbaixDiagrams"
            >
              <FormattedMessage
                defaultMessage="Pourbaix Diagrams"
                id="boilerplate.components.Header.pourbaixDiagrams"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/generalSearch"
            >
              <FormattedMessage
                defaultMessage="General Search"
                id="boilerplate.components.Header.generalSearch"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/publications"
            >
              <FormattedMessage
                defaultMessage="Publications"
                id="boilerplate.components.Header.publications"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
            <Styled(Link)
              to="/yourNextApp"
            >
              <FormattedMessage
                defaultMessage="Your Next App ..."
                id="boilerplate.components.Header.yourNextApp"
                tagName="span"
                values={Object {}}
              />
            </Styled(Link)>
          </styled.div>
        </styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
